src/routers/controllers/removeAuthorisedPersonRequestController.ts

```diff
@@ -0,0 +1,51 @@
+import { Request, Response } from "express";
+import * as constants from "../../constants";
+import { getExtraData, getLoggedInUserEmail, setExtraData } from "../../lib/utils/sessionUtils";
+import { Removal } from "../../types/removal";
+import { getCompanyAssociations, removeUserFromCompanyAssociations } from "../../services/associationsService";
+import { Associations } from "private-api-sdk-node/dist/services/associations/types";
+import logger from "../../lib/Logger";
+import { Session } from "@companieshouse/node-session-handler";
+import { CompanyNameAndNumber } from "../../types/util-types";
+import { validateRemoveAssociation } from "../../lib/validation/validateRemoveAssociation";
+
+/**
+ * This controller handles the removeUserFromCompanyAssociations (updateAssociationStatus) api call which
+ * updates the status for the association.
+ * If the user is removing themselves, it redirects them to the removing themselves success page, if removing
+ * another user it redirects to the manage authorised person page with a success message.
+ * If the request is invalid, the function throws an error
+ */
+export const removeAuthorisedPersonRequestController = async (req: Request, res: Response): Promise<void> => {
+
+    const companyNumber: string = req.params[constants.COMPANY_NUMBER];
+    const removal: Removal = getExtraData(req.session, constants.REMOVE_PERSON);
+
+    const companyAssociations: Associations = await getCompanyAssociations(req, companyNumber);
+
+    const associationToBeRemoved = companyAssociations.items.find(assoc => assoc.userEmail === removal.userEmail);
+
+    if (!associationToBeRemoved || !validateRemoveAssociation(associationToBeRemoved, removal, companyNumber)) {
```

Comment:
We're checking if `associationToBeRemoved` is `falsy` both outside, and inside the `validateRemoveAssociation` function - maybe we could remove it from the `validateRemoveAssociation` function and remove the valid `undefined` type from `foundAssociation`?  So users of the function know to check it outside?   Or we could remove the outside check?

Not too fussed as it's a very small detail but just a thought

Reply:
updated